### ISSUE TITLE
Python3.x encoding in headers and footers

### ISSUE THREAD

**Issue by austinorr** [elapouya/python-docx-template#46]
I am having problems with the regex in python3.x in the `DocxTemplate(object)` due to the function that searches for the xml encoding which begins on [line 99](https://github.com/elapouya/python-docx-template/blob/master/docxtpl/__init__.py#L99). In 3.x, the regex expression in `get_headers_footers_encoding(xml)` requires `xml` to be a string object, but it's getting passed as bytes and breaking the expression on [line 93](https://github.com/elapouya/python-docx-template/blob/master/docxtpl/__init__.py#L93). 

The traceback is 
`TypeError: can't use a string pattern on a bytes-like object`
python 3.5, 3.4
docxtpl 0.2.2
requirements as spec'd in a clean conda env

My quick workaround is inelegant but is working for python 3.x and shouldn't break the 2.7 compatibility.
The logic to assign the global is not shown, but the temporary workaround begins on line 101 of docxtpl/__init__.py:
```python
if PY3K:
    xml = xml.decode('utf-8')
encoding = self.get_headers_footers_encoding(xml)
if PY3K:
     xml = self.patch_xml(xml)
else:
     xml = self.patch_xml(xml).decode(encoding)
```

I assume your testing process includes 3.x and I see that headers and footers have given you plenty of headaches before, which means that it's likely working for you. Have you had problems with this regex traceback? If not, I wonder what going on with my env or package combination. Also, now that this is working for me it's seriously saving me tons of time, so thank you for writing/maintaining it. 
PS. The code above has not been run through your test suite, so caveat emptor to any others with the same problem that who hit this issue thread.

**Comment by elapouya** [comment]
Unfortunately, initially, I did not develop my module for python 3,
some helpped me with "six" module a long time ago,
it should this way to solve the problem.
I have to learn "six" before fixing PY3K issues...

**Comment by nostalgiaz** [comment]
+1

**Comment by nostalgiaz** [comment]
@elapouya have you already scheduled a release with py3 support?
Can we help you somehow?

**Comment by elapouya** [comment]
Hello,

Well, I do not use PY3K yet, so I cannot port docxtpl right now.

If you know the module 'six', you could help me to insert the right code to
have the lib working py2 and py3

2016-08-09 16:22 GMT+02:00 Mattia Larentis <notifications@github.com>:

**Comment by nostalgiaz** [comment]
@elapouya https://github.com/elapouya/python-docx-template/pull/47 
It's really simple :)

**Comment by elapouya** [comment]
fixed in 0.2.3 with your help
